cmake/compiler/gcc/target_base.cmake

```diff
@@ -18,3 +18,52 @@ macro(toolchain_cc_cstd_flag dest_var_name c_std)
   set_ifndef(${dest_var_name} "-std=${c_std}")
 
 endmacro()
+
+#
+# The two macros are dumping grounds for toolchain-specific options
+# that were previously in the top-level CMakeLists.txt. This is merely
+# a stopgap until the toolchain abstraction work underway (by others)
+# is complete. (There are two of these to preserve the original order.)
+#
+
+macro(toolchain_cc_others_1)
+  zephyr_cc_option(-fno-asynchronous-unwind-tables)
+  zephyr_cc_option(-fno-pie)
+  zephyr_cc_option(-fno-pic)
+  zephyr_cc_option(-fno-strict-overflow)
+
+  if(CONFIG_OVERRIDE_FRAME_POINTER_DEFAULT)
+    if(CONFIG_OMIT_FRAME_POINTER)
+      zephyr_cc_option(-fomit-frame-pointer)
+    else()
+      zephyr_cc_option(-fno-omit-frame-pointer)
+    endif()
+  endif()
+endmacro()
+
+macro(toolchain_cc_others_2)
+  if(NOT CMAKE_C_COMPILER_ID STREQUAL "Clang")
+    zephyr_cc_option(-fno-reorder-functions)
+
+    if(NOT ${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xcc")
+      zephyr_cc_option(-fno-defer-pop)
```

Comment:
Are the conditionals still needed? I mean we're now in a file named "something/gcc/something". Is that gcc name a lie? If yes a lie then worth some more details.

Conversely, did you just remove all other options from the Clang resp. xcc command lines?

Reply:
Yes, the conditionals are needed.  xcc, clang, etc. etc. (read: all the other "toolchains") mostly just include the files something/gcc/something.  If you look at the diffs side-by-side it's very clear that all I did was airlift these options from the top-level and put them in the only toolchain that matters.  Is this refactorization currently a giant mess?  Yes.  Is fixing that a problem for me or for this patch?  No.  Again, the only goal here is to get GCC-specific options out of the top-level CMakeLists.txt because the toolchain abstraction is completely pointless so long as the top-level remains compiler-specific.

Reply:
BTW, the proof that these macros exist on every path is obvious: cmake would simple refuse to configure if these macros aren't defined.  And they always end up defined for every platform checked by sanitycheck (even those that are build-only).  Thus cmake/compiler/gcc/target_base.cmake is included and the behavior before and after this patch is identical for all those platforms.

Reply:
> xcc, clang, etc. etc. (read: all the other "toolchains") mostly just include the files something/gcc/something.

Thx, I think this deserves to be stated that clearly the commit message (not sure about the "mostly" though). If "compiler-specific" were changed to "gcc-compatible" I think that would help to. Just pretend  for a minute that other people that the 3 or 5 who understand the toolchain "mess" might want to read this commit message.

> BTW, the proof that these macros exist on every path is obvious: cmake would simple refuse to configure if these macros aren't defined.

Sorry I'm not following, does CI test anything but the Zephyr SDK?

> every platform checked by sanitycheck (even those that are build-only).

I don't see anything platform or arch specific in this commit...?

BTW `zephyr_cc_option()` is "magical": it checks whether that option is supported or not and silently drops it when not. So I'm not sure how `zephyr_cc_option` can prove anything.

Reply:
> Thx, I think this deserves to be stated that clearly the commit message

What needs to be stated clearly in the commit message?

Reply:
> BTW `zephyr_cc_option()` is "magical": it checks whether that option is supported or not and silently drops it when not. So I'm not sure how `zephyr_cc_option` can prove anything.

Read again.  I did not say that `zephyr_cc_option()` proves anything.